### PR TITLE
SDD-762 course download

### PR DESCRIPTION
**Pull request by MrKevJoy**
- Updated page link to add a download button
- Added tests

### CODE REVIEW

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs [hunk 1]
```diff
@@ -52,9 +52,20 @@ public static ParsedCsvCourseRow FromCsvCourseRow(CsvCourseRow row, IEnumerable<
             _ => (CourseAttendancePattern?)null
         };
 
+        public static string ResolveAttendancePattern(CourseAttendancePattern value) => value switch
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:55]: I think we should use a prefix other than Resolve for these added methods that map to a string. The `*Resolved` fields are for the strongly-typed properties that have come _from_ a string. Perhaps `Map` (so `MapAttendancePattern`, `MapCost` etc)?

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs [hunk 2]
```diff
@@ -52,9 +52,20 @@ public static ParsedCsvCourseRow FromCsvCourseRow(CsvCourseRow row, IEnumerable<
...
+        {
+            CourseAttendancePattern.Daytime => "daytime",
+            CourseAttendancePattern.Evening => "evening",
+            CourseAttendancePattern.Weekend => "weekend",
+            CourseAttendancePattern.DayOrBlockRelease => "day/block release",
+            _ => null
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:61]: I think the `_` case should throw
```suggestion
            _ => throw new NotSupportedException($"Unknown value: '{value}'.")
```

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs [hunk 3]
```diff
@@ -65,9 +76,20 @@ public static ParsedCsvCourseRow FromCsvCourseRow(CsvCourseRow row, IEnumerable<
             _ => (CourseDeliveryMode?)null
         };
 
+        public static string ResolveDeliveryMode(CourseDeliveryMode value) => value switch
+        {
+            CourseDeliveryMode.ClassroomBased => "classroom based",
+            CourseDeliveryMode.Online => "online",
+            CourseDeliveryMode.WorkBased => "work based",
+            _ => null
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:84]:
```suggestion
            _ => throw new NotSupportedException($"Unknown value: '{value}'.")
```

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs [hunk 4]
```diff
@@ -78,6 +100,16 @@ public static ParsedCsvCourseRow FromCsvCourseRow(CsvCourseRow row, IEnumerable<
...
         };
 
+        public static string ResolveDurationUnit(CourseDurationUnit value) => value switch
+        {
+            CourseDurationUnit.Hours => "hours",
+            CourseDurationUnit.Days => "days",
+            CourseDurationUnit.Weeks => "weeks",
+            CourseDurationUnit.Months => "months",
+            CourseDurationUnit.Years => "years",
+            _ => null
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:110]:
```suggestion
            _ => throw new NotSupportedException($"Unknown value: '{value}'.")
```

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs [hunk 5]
```diff
@@ -104,6 +150,14 @@ public static ParsedCsvCourseRow FromCsvCourseRow(CsvCourseRow row, IEnumerable<
             _ => (CourseStudyMode?)null
         };
 
+        public static string ResolveStudyMode(CourseStudyMode value) => value switch
+        {
+            CourseStudyMode.FullTime => "full time",
+            CourseStudyMode.PartTime => "part time",
+            CourseStudyMode.Flexible => "flexible",
+            _ => null
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:158]:
```suggestion
            _ => throw new NotSupportedException($"Unknown value: '{value}'.")
```

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs (hunk 1 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs]: good shout, changed.

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs (hunk 2 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs]: changed

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs (hunk 3 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs]: changed

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs (hunk 4 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:110]: changed

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs (hunk 5 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs]: changed

**Review by gunndabad — APPROVED** [review]

**Pull request merged**